include/OpenColorIO/OpenColorIO.h

```diff
@@ -2714,13 +2676,55 @@ class OCIOEXPORT PlanarImageDesc : public ImageDesc
  * Inherit from the class to fully customize the implementation of a GPU shader program
  * from a color transformation.
  *
- * When no customizations are needed then the :cpp:class:`GpuShaderDesc` is a better choice.
+ * When no customizations are needed then the GpuShaderDesc is a better choice.
  *
  * To better decouple the DynamicProperties from their GPU implementation, the code provides
  * several addUniform() methods i.e. one per access function types. For example, an
  * ExposureContrastTransform instance owns three DynamicProperties and they are all
  * implemented by a double. When creating the GPU fragment shader program, the addUniform() with
  * GpuShaderCreator::DoubleGetter is called when property is dynamic, up to three times.
+ * 
+ * **An OCIO shader program could contain:**
+ *
+ * * A declaration part  e.g., uniform sampled3D tex3;
+ *
+ * * Some helper methods
+ *
+ * * The OCIO shader function may be broken down as:
+ *
+ *    * The function header  e.g., void OCIODisplay(in vec4 inColor) {
+ *    * The function body    e.g.,   vec4 outColor.rgb = texture3D(tex3, inColor.rgb).rgb;
+ *    * The function footer  e.g.,   return outColor; }
+ * 
+ * 
+ * **Usage Example:**
+ * 
+ * Below is a code snippet to highlight the different parts of the OCIO shader program.
+ * 
+ * \code{.cpp}
+ *     
+ *     // All global declarations
+ *     uniform sampled3D tex3;
+ *   
+ *     // All helper methods
+ *     vec3 computePosition(vec3 color)
+ *     {
+ *        vec3 coords = color;
+ *        // Some processing...
+ *        return coords;
+ *     }
+ *     
+ *     // The shader function
+ *     vec4 OCIODisplay(in vec4 inColor)     //
+ *     {                                     // Function Header
+ *        vec4 outColor = inColor;           //
+ *     
+ *        outColor.rgb = texture3D(tex3, computePosition(inColor.rgb)).rgb;
+ *     
+ *        return outColor;                   // Function Footer
+ *     }                                     //
+ * 
+ * \endcode
```

Comment:
These code examples show up on the Python side too.  
This is probably gonna complicate things, but is there any way we can write Python code that only shows up on the Python side, and cpp code that only shows up on the cpp side?

Reply:
This will require overriding the docstrings in question directly in the python binding source.